spec/presenters/post_presenter_spec.rb

```diff
@@ -20,6 +20,26 @@
     end
   end
 
+  describe "#with_interactions" do
+    it "works with a user" do
+      expect(@presenter.with_interactions).to be_a Hash
+    end
+
+    it "works without a user" do
+      expect(@unauthenticated_presenter.with_interactions).to be_a Hash
+    end
+  end
+
+  describe "#with_initial_interactions" do
+    it "works with a user" do
+      expect(@presenter.with_initial_interactions).to be_a Hash
+    end
```

Comment:
I think these tests could be a little smarter and verify the contents of the hash. This data isn't checked anywhere now.